backend/api.test/Client/MissionTests.cs

```diff
@@ -0,0 +1,432 @@
+                JsonSerializer.Serialize(query),
+                null,
+                "application/json"
+            );
+
+            // Increasing pageSize to 50 to ensure the missions we are looking for is included
+            string urlMissionRuns = "/missions/runs?pageSize=50";
+            var response = await _client.GetAsync(urlMissionRuns);
+            var missionRuns = await response.Content.ReadFromJsonAsync<List<MissionRun>>(
+                _serializerOptions
+            );
+            Assert.True(response.IsSuccessStatusCode);
+            Assert.True(missionRuns != null);
+            int missionRunsBefore = missionRuns.Count;
+
+            await _client.PostAsync(missionsUrl, content);
+            await _client.PostAsync(missionsUrl, content);
+            await _client.PostAsync(missionsUrl, content);
+
+            response = await _client.GetAsync(urlMissionRuns);
+            missionRuns = await response.Content.ReadFromJsonAsync<List<MissionRun>>(
+                _serializerOptions
+            );
+
+            // Assert
+            Assert.True(response.IsSuccessStatusCode);
+            Assert.True(missionRuns != null);
+            Assert.True(missionRuns.Count == (missionRunsBefore + 3));
+        }
+
+        [Fact]
+        public async Task GetMissionById_ShouldReturnNotFound()
+        {
+            string missionId = "RandomString";
+            string url = "/missions/runs/" + missionId;
+            var response = await _client.GetAsync(url);
+            Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
+        }
+
+        [Fact]
+        public async Task DeleteMission_ShouldReturnNotFound()
+        {
+            string missionId = "RandomString";
+            string url = "/missions/runs/" + missionId;
+            var response = await _client.DeleteAsync(url);
+            Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
+        }
+
+        [Fact]
+        public async Task StartMissionTest()
+        {
+            // Arrange - Robots
+            string robotUrl = "/robots";
+            string missionsUrl = "/missions";
+            var response = await _client.GetAsync(robotUrl);
+            Assert.True(response.IsSuccessStatusCode);
+            var robots = await response.Content.ReadFromJsonAsync<List<Robot>>(_serializerOptions);
+            Assert.True(robots != null);
+            var robot = robots[0];
+            string robotId = robot.Id;
+
+            // Arrange - Areas
+            string areaUrl = "/areas";
+            var areaResponse = await _client.GetAsync(areaUrl);
+            Assert.True(areaResponse.IsSuccessStatusCode);
+            var areaResponses = await areaResponse.Content.ReadFromJsonAsync<List<AreaResponse>>(_serializerOptions);
+            Assert.True(areaResponses != null);
+            var area = areaResponses[0];
+            string areaName = area.AreaName;
+            string installationCode = area.InstallationCode;
```

Comment:
Is it possible / would it make sense to reuse some code here? I see this snippet is recurring a few places

Reply:
It is possible yes, but I was not if I wanted to do that or not, to make the tests as atomic as possible and to avoid the problem we had with the PopulateAreaDb where several tests tried to add the same Area to the DB and we got unique constraint conflicts. But I am open to ideas.

Reply:
I see, good enough reason for me